keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -23,21 +21,25 @@ data class Resource<T : ResourceSpec>(
 
     other as Resource<*>
 
-    if (apiVersion != other.apiVersion) return false
     if (kind != other.kind) return false
     if (spec != other.spec) return false
 
     return true
   }
 
   override fun hashCode(): Int {
-    var result = apiVersion.hashCode()
-    result = 31 * result + kind.hashCode()
+    var result = kind.hashCode()
     result = 31 * result + spec.hashCode()
     return result
   }
 }
 
+/**
+ * Creates a resource id in the correct format.
+ */
+fun generateId(kind: ResourceKind, spec: ResourceSpec) =
+  "${kind.group}:${kind.kind}:${spec.id}"
```

Comment:
When it comes time to assign an id to a resource I'm using the group (e.g. `ec2`) and the un-qualified kind (e.g. `cluster`). I do _not_ want the version to be part of the id as we should consider a `ec2/cluster@v1` and an `ec2/cluster@v2` with the same moniker as a conflict (because they're specifying the same actual cluster in the cloud).